### PR TITLE
Enables relays 1-4 on Beaglebone Blue GP0 connector

### PR DESCRIPTION
**Pull request by TritonSailor**
This pull request enables relay functionality on Beaglebone Blue.  It maps ardupilot relays 1-4 to the Beaglebone Blue GP0 connector pins GP0_3-6.

### CODE REVIEW

File: libraries/AP_Relay/AP_Relay.cpp [hunk 1]
```diff
@@ -25,10 +25,19 @@
 #elif CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN
 #define RELAY1_PIN_DEFAULT 33
 #define RELAY2_PIN_DEFAULT -1
+#elif CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+#ifdef HAL_BOARD_SUBTYPE_LINUX_BLUE
```
Comment (khancyr) [inline libraries/AP_Relay/AP_Relay.cpp]: #if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BLUE

File: libraries/AP_Relay/AP_Relay.cpp [hunk 2]
```diff
@@ -25,10 +25,19 @@
...
+#define RELAY1_PIN_DEFAULT 57
+#define RELAY2_PIN_DEFAULT 49
+#define RELAY3_PIN_DEFAULT 116
+#define RELAY4_PIN_DEFAULT 113
+#endif
 #else
 // no relay for this board
 #define RELAY1_PIN_DEFAULT -1
 #define RELAY2_PIN_DEFAULT -1
+#define RELAY3_PIN_DEFAULT -1
```
Comment (khancyr) [inline libraries/AP_Relay/AP_Relay.cpp]: you need to add this to all boards ! otherwise it will be missing

Comment (OXINARF) [reply libraries/AP_Relay/AP_Relay.cpp]: Better to remove the else and do a block of

```
#ifndef RELAY1_PIN_DEFAULT
  #define RELAY1_PIN_DEFAULT -1
#endif

[...]
```
Then all the above defining -1 can be removed.

Comment (khancyr) [reply libraries/AP_Relay/AP_Relay.cpp]: agree

Comment (TritonSailor) [reply libraries/AP_Relay/AP_Relay.cpp]: The above suggestion is now implemented and confirmed working on the Blue.  I'll cleanup the PR and push the changes later today...

**Comment by mirkix** [comment]
@TritonSailor Thanks for your PR!

**Comment by mirkix** [comment]
@OXINARF LGTM

**Comment by OXINARF** [comment]
I made some small corrections (small indentation issues and use of ifdef for subtype instead of comparing to subtype macro, see https://github.com/ArduPilot/ardupilot/commit/35dddbd35aa8af7d65bfd1e8b3cddb9194ccc833#diff-796c2145f6c474425239250e7dc75990R27) and merged.

Thank you for your first contribution @TritonSailor!

**Pull request closed without merging**